library_api/controllers/book_controller.go

```diff
@@ -0,0 +1,75 @@
+package controllers
+
+import (
+	"encoding/json"
+	"fmt"
+	"library_api/entities"
+	"library_api/repositories"
+	"net/http"
+	"sort"
+	"strconv"
+)
+
+func CreateBooks(w http.ResponseWriter, r *http.Request) { // TODO discuss about create action
+	name := r.FormValue("name")
+	author := r.FormValue("author")
+	year, err := strconv.Atoi(r.FormValue("year"))
+
+	if err != nil {
+		fmt.Println("year incorrect")
+		w.WriteHeader(http.StatusBadRequest)
+		panic(err)
+	}
+
+	if name == "" || author == "" {
+		fmt.Println("name or author incorrect")
+		w.WriteHeader(http.StatusBadRequest)
+	} else {
+		books := []entities.Book{
+			{name, author, year},
+		}
+
+		book_repository := repositories.NewBookRepository()
+
+		_, err := book_repository.Create(books)
+		if err != nil {
+			panic(err)
+		}
+
+		fmt.Println("Successfully Create books")
+		w.WriteHeader(http.StatusCreated)
+	}
+}
+
+func ListOfBooks(w http.ResponseWriter, r *http.Request) {
```

Comment:
when you assign your controllers as a methods to Book struct you can rename it just `List`